satellite/audit/verifier_test.go

```diff
@@ -466,83 +534,97 @@ func TestVerifierDeletedSegment(t *testing.T) {
 	testplanet.Run(t, testplanet.Config{
 		SatelliteCount: 1, StorageNodeCount: 4, UplinkCount: 1,
 	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
-		audits := planet.Satellites[0].Audit.Service
-		err := audits.Close()
-		require.NoError(t, err)
+		planet.Satellites[0].Audit.Worker.Loop.Pause()
+
+		audits := planet.Satellites[0].Audit
+		queue := audits.Queue
+		satellite := planet.Satellites[0]
 
 		ul := planet.Uplinks[0]
 		testData := testrand.Bytes(8 * memory.KiB)
 
-		err = ul.Upload(ctx, planet.Satellites[0], "testbucket", "test/path", testData)
+		err := ul.Upload(ctx, planet.Satellites[0], "testbucket", "test/path", testData)
 		require.NoError(t, err)
 
-		stripe, _, err := audits.Cursor.NextStripe(ctx)
+		satellite.Audit.Chore.Loop.TriggerWait()
+		path, err := queue.Next()
 		require.NoError(t, err)
 
 		// delete the file
 		err = ul.Delete(ctx, planet.Satellites[0], "testbucket", "test/path")
 		require.NoError(t, err)
 
-		report, err := audits.Verifier.Verify(ctx, stripe, nil)
+		report, err := audits.Verifier.Verify(ctx, path, nil)
 		require.True(t, audit.ErrSegmentDeleted.Has(err))
 		assert.Empty(t, report)
 	})
 }
 
-func TestVerifierModifiedSegment(t *testing.T) {
-	testplanet.Run(t, testplanet.Config{
-		SatelliteCount: 1, StorageNodeCount: 4, UplinkCount: 1,
-	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
-		audits := planet.Satellites[0].Audit.Service
-		err := audits.Close()
-		require.NoError(t, err)
-
-		ul := planet.Uplinks[0]
-		testData := testrand.Bytes(8 * memory.KiB)
+// TODO: Commenting this test for now because currently when we get paths from the audit queue,
```

Comment:
We need this for correctness, so it must not be removed.

Reply:
okay. how do you think we should access the old pointer?

Reply:
Before starting the audit, you load it from metainfo.

Reply:
I'm not sure I totally know where you mean. This `checkIfSegmentDeleted` function is called early on in the `Verify` function:
```
// checkIfSegmentDeleted checks if path's pointer has been deleted since path was selected.
func (verifier *Verifier) checkIfSegmentDeleted(ctx context.Context, segmentPath string) (newPointer *pb.Pointer, err error) {
	defer mon.Task()(&ctx)(&err)

	newPointer, err = verifier.metainfo.Get(ctx, segmentPath)
	if err != nil {
		if storage.ErrKeyNotFound.Has(err) {
			return nil, ErrSegmentDeleted.New("%q", segmentPath)
		}
		return nil, err
	}

	// TODO: commenting because we no longer have access to the old pointer
	//if oldPointer != nil && oldPointer.CreationDate != newPointer.CreationDate {
	//	return nil, ErrSegmentDeleted.New("%q", segmentPath)
	//}
	return newPointer, nil
}
```
Should a metainfo.Get call happen before this call in `Verify` and that will be the old pointer? And then inside this function we call metainfo.Get again, and this time that will be the `newPointer`?

Reply:
It could happen at the start of verify or before verify, but we need this logic of:

1. load pointer from metainfo
2. audit nodes
3. check whether pointer in metainfo still matches step 1.
4. handle audit failures / successes

But, yes, `checkIfSegmentDeleted` is probably not the best place to load the old pointer.

Reply:
the way audit service 1.0 worked though, it didn't look like it followed those steps either, since this comparison between old and new pointers happened before auditing the nodes occurred right?

Reply:
In https://github.com/storj/storj/blob/master/satellite/audit/verifier.go#L109 the check happens after downloading the pieces, only then it goes to check the downloaded pieces. The old pointer was already in the initial call.

Reply:
ah okay I see. do you have any opinion then on where to load the old pointer? should it still happen in the top of the verify call?